### DISCUSSION TITLE
Preprocessing of amino acid sequences

### DISCUSSION THREAD

**Discussion by TrikiSadok** [rdkit/rdkit#3784] [Q&A]
I am not that familiar with RDKit. 
Does RDKIt allow transforming amino acid sequences into numerical features for machine learning purposes?
example of amino acid sequence I am trying to transform into numerical descriptor that is included in the data set I have: 
"
MIGIDIVSIARVEKCVKRFEMRFLERFLSPSEIVLCKDKSSSIAGFFALKEACSKALQVGIGKELSFLDMRISKSPKNAPLITLSKEKMDYFNIQSLSASISHDAGFAIAVVMISP
"
Thanks in advance for any help.

**Comment by greglandrum** [comment]
It does! You can convert a sequence into an RDKit molecule as follows:
```
>>> m = Chem.MolFromSequence('MIGIDIVSIARVEKCVKRFEMRFLERFLSPSEIVL')
```
And then all the standard RDKit descriptors can be used:
```
>>> from rdkit.Chem import rdMolDescriptors
>>> rdMolDescriptors.CalcNumAmideBonds(m)
34
```

**Reply by TrikiSadok** [reply]
Thanks a lot.
Is there a difference in treatment between MolFromSequence() and MolFromFASTA() in this case ?? since I found in other forums people are using MolFromFASTA() to transform amino acid and protein sequences.

**Reply by greglandrum** [reply]
No. Other than the obvious fact that a different parser is used for the different formats, the molecules are processed the same way.

**Reply by TrikiSadok** [reply]
very clear, Thanks a lot.
